authors/apps/authentication/views.py

```diff
@@ -198,17 +198,17 @@ def post(self, request):
         # Query for email in database
         user = User.objects.filter(email=request.data['email']).first()
         if user is None:
-            return Response({"message": "The email you entered does not exist"})
+            return Response({"message": "The email you entered does not exist"}, status=status.HTTP_404_NOT_FOUND)
 
         # Get URL for client and include in the email for resetting password
-        current_site_domain = settings.CLIENT_URL
+        current_site_domain = request.META['HTTP_ORIGIN']
         # generate token
         token = default_token_generator.make_token(user)
 
         # Sends mail with url, path of reset password and token
         mail_message = 'Dear ' + user.username + ',\n\n''We received a request to change your password on Authors Haven.\n\n' \
                                                  'Click the link below to set a new password' \
-                                                 ' \n' + current_site_domain + '/reset-password/' + token + '/' \
+                                                 ' \n' + current_site_domain + '/reset-password/?' + token + '/'+user.email + \
             '\n\nYours\n AuthorsHaven.'
         SendMail(subject="Reset Password",
```

Comment:
During our demo,PO said there is no need to prompt users for their email because they are redirected from their email address. On the implementation side when users are setting the new password, we require the email address to query the user from the database. When generatingthe token,we have used username so user email is not incoporated in the token. Let me know your take.

Reply:
I have included the email so that I can capture it from the URL so that I include it in the payload I send using react

Reply:
It is much cleaner and safer to use query parameters correctly. And also instead of concatenating strings make use of the .format() to add variables to the message strings.

Reply:
Okay

Reply:
Check it now and let me know whether it meets and exceeds your expectations